src/main/kotlin/no/skatteetaten/aurora/boober/feature/ToxiproxySidecarSpec.kt

```diff
@@ -68,4 +90,107 @@ fun List<Container>.overrideEnvVarsWithProxies(adc: AuroraDeploymentSpec, contex
+    groupToxiproxyServerAndPortFields().keys.toList()
+
+fun AuroraDeploymentSpec.getToxiproxyNames(): List<String> =
+    getToxiproxyEndpointNames() + getToxiproxyServerAndPortNames()
+
+// Validate that for every endpoint in toxiproxy/endpointsFromConfig, there is a corresponding environment variable
+fun AuroraDeploymentSpec.missingEndpointVariableErrors() =
+    groupToxiproxyEndpointFields().keys.mapNotNull { varName ->
+        if (getSubKeys("config").keys.none { it.removePrefix("config/") == varName }) {
+            AuroraDeploymentSpecValidationException(
+                "Found Toxiproxy config for endpoint named $varName, " +
+                    "but there is no such environment variable."
+            )
+        } else null
+    }
+
+// Validate that the environment variables given in toxiproxy/serverAndPortFromConfig exist
+fun AuroraDeploymentSpec.missingServerAndPortVariableErrors() =
+    groupToxiproxyServerAndPortFields().flatMap { (proxyName, fields) ->
+        listOf("server", "port").mapNotNull { serverOrPort ->
+            val serverOrPortVariable = fields
+                .find { it.key == "toxiproxy/serverAndPortFromConfig/$proxyName/${serverOrPort}Variable" }!!
+                .value
+                .value<String>()
+            if (getSubKeys("config").keys.none { it.removePrefix("config/") == serverOrPortVariable }) {
+                AuroraDeploymentSpecValidationException(
+                    "Found Toxiproxy config for a $serverOrPort variable named $serverOrPortVariable, " +
+                        "but there is no such environment variable."
+                )
+            } else null
+        }
+    }
+
+// Validate that there are no proxyname duplicates
+fun AuroraDeploymentSpec.proxynameDuplicateErrors() = getToxiproxyNames()
+    .groupingBy { it }
+    .eachCount()
+    .filter { it.value > 1 }
+    .map {
+        AuroraDeploymentSpecValidationException(
+            "Found ${it.value} Toxiproxy configs with the proxy name \"${it.key}\". " +
+                "Proxy names have to be unique."
+        )
+    }
+
+fun MutableList<ToxiproxyConfig>.addEndpointsFromConfigAndReturnNextPort(
+    spec: AuroraDeploymentSpec,
+    initialPort: Int
+): Int {
+    var port = initialPort
+    spec.extractToxiproxyEndpoints().forEach { (proxyname, varname) ->
+        val url = spec.fields["config/$varname"]?.value<String>()
+        if (url != null) {
+            val uri = URI(url)
+            val upstreamPort = if (uri.port == -1) {
+                if (uri.scheme == "https") PortNumbers.HTTPS_PORT else PortNumbers.HTTP_PORT
+            } else uri.port
+            add(
+                ToxiproxyConfig(
+                    name = proxyname,
+                    listen = "0.0.0.0:$port",
+                    upstream = uri.host + ":" + upstreamPort
+                )
+            )
+            port++
+        }
+    }
+    return port
+}
+
+fun MutableList<ToxiproxyConfig>.addServersAndPortsFromConfig(
```

Comment:
Dette begynner å bli veldig småpirk, men jeg hadde nok foretrukket at denne ikke muterte listen men heller returnerte en liste. kunne man brukt `map` i stedet for `forEach`?
jeg godkjenner, så får du vurdere om du ønsker å gjøre noe med det

Reply:
Jeg skal se på det :-)

Reply:
Har endret på det nå. Dette blir litt kronglete på grunn av at alle proxyene må få unike portnumre, men jeg har laget en løsning på det. La også til en liten enhetstest for å forsikre at portlogikken fungerer som forventet.